ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LerpExtensions.cs

```diff
@@ -0,0 +1,151 @@
+
+			Vector3 point1 = path[index1];
+			Vector3 point2 = path[index2];
+			return Vector3.Lerp(point1, point2, t - index1);
+		}
+
+		public static void LerpTransforms(this List<Transform> path, float t, Transform outVal)
+		{
+			Debug.Assert(path != null, "Path is null");
+			Debug.Assert(path.Count > 0, "Path is empty");
+
+			t = Mathf.Clamp01(t);
+
+			int index1 = Mathf.FloorToInt(t * (path.Count - 1));
+			int index2 = Mathf.CeilToInt(t * (path.Count - 1));
+
+			index1 = Mathf.Clamp(index1, 0, path.Count - 1);
+			index2 = Mathf.Clamp(index2, 0, path.Count - 1);
+
+			float segmentT = t * (path.Count - 1) - index1;
+
+			Transform point1 = path[index1];
+			Transform point2 = path[index2];
+			outVal.SetPositionAndRotation(
+				Vector3.Lerp(point1.position, point2.position, segmentT),
+				Quaternion.Lerp(point1.rotation, point2.rotation, segmentT)
+				);
+			outVal.localScale = Vector3.Lerp(point1.localScale, point2.localScale, segmentT);
+		}
+		public static void LerpTransforms(this Transform[] path, float t, Transform outVal)
+		{
+			Debug.Assert(path != null, "Path is null");
+			Debug.Assert(path.Length > 0, "Path is empty");
+
+			t = Mathf.Clamp01(t);
+
+			int index1 = Mathf.FloorToInt(t * (path.Length - 1));
+			int index2 = Mathf.CeilToInt(t * (path.Length - 1));
+
+			index1 = Mathf.Clamp(index1, 0, path.Length - 1);
+			index2 = Mathf.Clamp(index2, 0, path.Length - 1);
+
+			float segmentT = t * (path.Length - 1) - index1;
+
+			Transform point1 = path[index1];
+			Transform point2 = path[index2];
+			outVal.SetPositionAndRotation(
+				Vector3.Lerp(point1.position, point2.position, segmentT),
+				Quaternion.Lerp(point1.rotation, point2.rotation, segmentT)
+				);
+			outVal.localScale = Vector3.Lerp(point1.localScale, point2.localScale, segmentT);
+		}
+
+		public static Vector3 LerpPositions(this List<Transform> path, float t)
+		{
+			Debug.Assert(path != null, "Path is null");
+			Debug.Assert(path.Count > 0, "Path is empty");
+
+			t = Mathf.Clamp01(t);
+
+			int index1 = Mathf.FloorToInt(t * (path.Count - 1));
+			int index2 = Mathf.CeilToInt(t * (path.Count - 1));
+
+			index1 = Mathf.Clamp(index1, 0, path.Count - 1);
+			index2 = Mathf.Clamp(index2, 0, path.Count - 1);
+
+			Vector3 point1 = path[index1].position;
+			Vector3 point2 = path[index2].position;
+			return Vector3.Lerp(point1, point2, t - index1);
+		}
+		public static Vector3 LerpPositions(this Transform[] path, float t)
+		{
+			Debug.Assert(path != null, "Path is null");
+			Debug.Assert(path.Length > 0, "Path is empty");
+
+			t = Mathf.Clamp01(t);
+
+			int index1 = Mathf.FloorToInt(t * (path.Length - 1));
+			int index2 = Mathf.CeilToInt(t * (path.Length - 1));
+
+			index1 = Mathf.Clamp(index1, 0, path.Length - 1);
+			index2 = Mathf.Clamp(index2, 0, path.Length - 1);
+
+			Vector3 point1 = path[index1].position;
+			Vector3 point2 = path[index2].position;
+			return Vector3.Lerp(point1, point2, t - index1);
+		}
```

Comment:
I think LerpQuaternions should be here as well

Reply:
Resolved with commit 3d3a171e473e0cc88d49b8524ab9817afe1532e2